### PR TITLE
I want a Semantic Token Provider for the VSCode Extension

### PR DESCRIPTION
**Pull request by cvchaparro**
As an AaC VSCode user, I want a Semantic Token Provider for the VSCode Extension, so that the AaC VSCode plugin can provide dynamically registered keywords, types, and enums.

## References:

https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide

Tokens are classified into types noted here: https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#standard-token-types-and-modifiers

## AC:
- [x] The LSP supports the `DocumentSemanticTokensProvider` 
- [x] The VSCode Extension provides a dynamic list of keywords for the currently defined root keys as `namespace` tokens.
- [x] The VSCode Extension provides a dynamic list of types based on the currently defined definitions' names as `class` tokens.
- [x] The VScode Extension provides a dynamic list of enums based enum-defined values as `enumMember`
- [x] The VScode Extension provides a dynamic list of field names such as `name:` or `type:` based on the declaration of schema-defined field names as `struct` (maybe an alt?)

### CODE REVIEW

**Comment by cvchaparro** [comment]
I did some testing with the Scope/Token inspector and after some refactoring and clean up it looks like VSCode isn't really sending the cancellation requests anymore.

File: docs/docs/vscode_extension/lsp_features.md [hunk 1]
```diff
@@ -123,3 +123,17 @@ and press "Enter".
...
+experience with customized syntax coloring for specific AaC items. Without
+semantic highlighting, all AaC definitions will look the same like standard YAML
+but with semantic highlighting observant users can distinguish various types of
+definitions by their syntax coloring. For example, depending on your chosen
+theme, an enum definition will have one color compared to a schema definition,
+and so on.
+
+
+![TODO: Add image showing no semantic highlighting]()
+![TODO: Add image showing semantic highlighting]()
```
Comment (cvchaparro) [inline docs/docs/vscode_extension/lsp_features.md:139]: Still need to add some images for documentation

File: vscode_extension/src/aacExecutableWrapper.ts [hunk 2]
```diff
@@ -65,20 +65,24 @@ export async function getAaCVersion(): Promise<string | null> {
 }
 
 async function getCommandArgUserInput(commandArguments: CommandArgument[]) {
+    const hasPathArgument = (argument: CommandArgument, options: string[]) =>
```
Comment (cvchaparro) [inline vscode_extension/src/aacExecutableWrapper.ts:68]: Possibly handle selecting multiple files/directories

File: python/tests/helpers/lsp/responses/semantic_tokens_response.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from tests.helpers.lsp.responses.lsp_response import LspResponse
+
+
+class SemanticTokensResponse(LspResponse):
+    def get_default_data(self) -> list[int]:
+        return self.response.get("data", [])
+
+    def get_semantic_tokens(self) -> list[list[int]]:
+        data = self.get_default_data()
+        return [data[i:i + 5] for i in range(0, len(data), 5)]
```
Comment (Coffee2Bits) [inline python/tests/helpers/lsp/responses/semantic_tokens_response.py:10]: Out of curiosity, what's the '5' for? Should this be dynamic or configurable?

**Review by Coffee2Bits — APPROVED** [review]
Fantastic work! There are some clashes with some changes I made to the parser, but I'll resolve those on my branch.

File: python/tests/helpers/lsp/responses/semantic_tokens_response.py (hunk 3 above)
Comment (cvchaparro) [reply python/tests/helpers/lsp/responses/semantic_tokens_response.py:10]: Each semantic token is represented by 5 integers - the line number relative to the last token, the start character relative to the last token, the length of the token, the index corresponding to the token type in the `SemanticTokensLegend`, and a bitmask of all corresponding token modifiers in the `SemanticTokensLegend`.

For reference see the section on [Integer Encoding for Tokens](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens:~:text=relative%27%3B-,Integer%20Encoding%20for%20Tokens,-On%20the%20capability)

**Pull request merged**